tools/run_tests/python_utils/start_port_server.py

```diff
@@ -111,15 +112,15 @@ def start_port_server(port_server_port):
       try:
         urllib.request.urlopen('http://localhost:%d/get' % port_server_port,
                         timeout=1).read()
-        print('port server is up and ready')
+        logging.info('port server is up and ready')
         break
       except socket.timeout:
-        print('waiting for port_server: timeout')
+        logging.exception('waiting for port_server: timeout')
         traceback.print_exc();
         time.sleep(1)
         waits += 1
       except urllib.error.URLError:
-        print('waiting for port_server: urlerror')
+        logging.exception('waiting for port_server: urlerror')
         traceback.print_exc();
```

Comment:
Here the "urlerror" bit is arguably redundant. Otherwise the custom message provides context as a preamble. Done.